torch/autograd/_functions/blas.py

```diff
@@ -192,10 +192,3 @@ def backward(self, grad_output):
             grad_vector2 = vector1.mul(grad_output[0])
 
         return grad_vector1, grad_vector2
-
-
-# TODO: cross
-# TODO: diag
-# TODO: trace
-# TODO: tril
-# TODO: triu
```

Comment:
yes they are implemented, see [here](https://github.com/pytorch/pytorch/blob/master/torch/autograd/variable.py#L682,L689) and [here](https://github.com/pytorch/pytorch/blob/master/torch/autograd/_functions/linalg.py#L6,L42)